aws-rds-dbsecuritygroup/src/main/java/software/amazon/rds/dbsecuritygroup/CreateHandler.java

```diff
@@ -0,0 +1,72 @@
+package software.amazon.rds.dbsecuritygroup;
+
+import java.util.Optional;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.awssdk.services.rds.model.DbSecurityGroupAlreadyExistsException;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.HandlerErrorCode;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    public static final int PHYSICAL_RESOURCE_ID_MAX_LENGTH = 255;
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger) {
+
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+                .then(progress -> {
+                    final ResourceModel model = progress.getResourceModel();
+                    if (StringUtils.isNullOrEmpty(model.getGroupName())) {
+                        model.setGroupName(
+                                IdentifierUtils.generateResourceIdentifier(
```

Comment:
I would check that the format of this generated name matches existing resources. We've found issues in the past where customers relied on presence of stack-id in the primary identifier.